// Copyright 2021 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";

message SourcerepoRepo {
  string name = 1;
  int64 size = 2;
  string url = 3;
  repeated SourcerepoRepoPubsubConfigs pubsub_configs = 4;
  string project = 5;
}

message SourcerepoRepoPubsubConfigs {
  string topic = 1;
  string message_format = 2;
  string service_account_email = 3;
}

message ApplySourcerepoRepoRequest {
  SourcerepoRepo resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteSourcerepoRepoRequest {
  string service_account_file = 1;
  SourcerepoRepo resource = 2;
}

message ListSourcerepoRepoRequest {
  string service_account_file = 1;
  string Project = 2;
}

message ListSourcerepoRepoResponse {
  repeated SourcerepoRepo items = 1;
}

service SourcerepoRepoService {
  rpc ApplySourcerepoRepo(ApplySourcerepoRepoRequest) returns (SourcerepoRepo);
  rpc DeleteSourcerepoRepo(DeleteSourcerepoRepoRequest) returns (google.protobuf.Empty);
  rpc ListSourcerepoRepo(ListSourcerepoRepoRequest) returns (ListSourcerepoRepoResponse);
}
